src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -35,13 +36,14 @@ public class DefaultHtmlGenerator : IHtmlGenerator
         /// </summary>
         public DefaultHtmlGenerator(
             [NotNull] AntiForgery antiForgery,
-            [NotNull] IScopedInstance<ActionBindingContext> bindingContextAccessor,
+            [NotNull] IOptions<MvcOptions> optionsAccessor,
             [NotNull] IModelMetadataProvider metadataProvider,
             [NotNull] IUrlHelper urlHelper,
             [NotNull] IHtmlEncoder htmlEncoder)
         {
             _antiForgery = antiForgery;
-            _bindingContextAccessor = bindingContextAccessor;
+            var clientValidatorProviders = optionsAccessor.Options.ClientModelValidatorProviders.ToArray();
```

Comment:
sorry, my comment wasn't clear.  my complaint was about the `ToArray()` call: `DefaultHtmlGenerator` looks at `ClientModelValidatorProviders` once and calls the `CompositeClientModelValidatorProvider` once.  so if there's a (perf) reason for `ToArray()`, the concern is within `CompositeClientModelValidatorProvider`.  likely don't need it at all.